### PR TITLE
chore(angular): ast utils for angular with nx devkit

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior

AST Utils for Angular relied on the Angular Devkit API

## Expected Behavior

AST Utils for Angular  should have an Nx Devkit option

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/workspace/src/utilities/cli-config-utils.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { NxJsonConfiguration, readJson, Tree } from '@nrwl/devkit';
+
+export function getNpmScope(host: Tree): string {
```
Comment (leosvelperez) [inline packages/workspace/src/utilities/cli-config-utils.ts:3]: This one shouldn't be needed, we can use the `getWorkspaceLayout` from the Nx DevKit.

**Review by leosvelperez — CHANGES_REQUESTED** [review]

File: packages/angular/src/utils/nx-devkit/ast-utils.ts [hunk 2]
```diff
@@ -0,0 +1,629 @@
+import * as ts from 'typescript';
+import {
+  findNodes,
+  getImport,
+  getSourceNodes,
+} from '@nrwl/workspace/src/utils/ast-utils';
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/utils/nx-devkit/ast-utils.ts:6]: These imports are agnostic of the Devkit running them though. 

I don't mind updating the imports to be more specific if there is an understanding that `utils` should be legacy / deprecated.

File: packages/devkit/src/utils/move-dir.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Tree } from '@nrwl/tao/src/shared/tree';
+
+export function moveFilesToNewDirectory(
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/devkit/src/utils/move-dir.ts:3]: Great idea!

File: packages/workspace/src/utilities/ast-utils.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
+import { NxJsonConfiguration, readJson, Tree } from '@nrwl/devkit';
+import * as ts from 'typescript';
+import { findNodes } from './typescript/find-nodes';
+
+export function libsDir(host: Tree) {
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/workspace/src/utilities/ast-utils.ts:5]: Perfect!
I wasn't aware of that utility 🤦‍♂️

File: packages/workspace/src/utilities/cli-config-utils.ts (hunk 1 above)
Comment (Coly010) [reply packages/workspace/src/utilities/cli-config-utils.ts:3]: Awesome!

**Review by leosvelperez — APPROVED** [review]
LGTM! 🎉

**Pull request merged**